File: timesheet-service/src/main/java/com/wardlog/timesheetservice/enums/ActivityType.java
package com.wardlog.timesheetservice.enums;

public enum ActivityType {

    clinicblock("clinic block"),
    oncall("on call"),
    onsiteoncall("on site on call"),
    surgeryblock("surgery block");

    private final String label;

    ActivityType(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}

File: frontend/src/lib/activityType.ts
import type { ActivityType } from '../types/timesheet'

interface ActivityTypeOption {
  value: ActivityType
  label: string
  colorToken: string
}

export const ACTIVITY_TYPE_OPTIONS: ActivityTypeOption[] = [
  { value: 'CLINIC_BLOCK', label: 'Clinic Block', colorToken: 'clinic' },
  { value: 'SURGERY_BLOCK', label: 'Surgery Block', colorToken: 'surgery' },
  { value: 'ON_CALL', label: 'On-Call', colorToken: 'oncall' },
  { value: 'ON_SITE_ON_CALL', label: 'Onsite On-Call', colorToken: 'onsite-oncall' },
]
